base/linalg/generic.jl

```diff
@@ -870,8 +870,9 @@ true
 """
 function istriu(A::AbstractMatrix)
     m, n = size(A)
+    z = zero(eltype(A))
     for j = 1:min(n,m-1), i = j+1:m
-        if A[i,j] != 0
+        if A[i,j] != z
```

Comment:
Use `A[i,j] != zeros(A[i,j])` for the same reason as before and since the element type could potentially change over the array. This would also fail if the array was a `Vector{Any}`.